examples/plotting/file/custom_datetime_axis.py

```diff
@@ -0,0 +1,65 @@
+from pdb import set_trace as bp
+from math import pi
+
+import pandas as pd
+
+from bokeh.sampledata.stocks import MSFT
+from bokeh.plotting import figure, show, output_file
+from bokeh.models.formatters import TickFormatter, String, List
+
+# In this custom TickFormatter, xaxis labels are taken from an array of date
+# Strings (e.g. ['Sep 01', 'Sep 02', ...]) passed to the date_labels property. 
+class DateGapTickFormatter(TickFormatter):
+  date_labels = List(item_type=String)
+  __implementation__ = """
```

Comment:
1. Indentation is off, should be 4 spaces.
2. For readability, one empty line before `__implementation__` should be added.
3. `item_type` is unnecessary, `List(String)` is the proper style (which should resemble scala's `List[String]` syntax).